src/gui/Gui.cc

```diff
@@ -148,9 +148,23 @@ std::unique_ptr<ignition::gui::Application> createGui(
     return nullptr;
 
   std::size_t runnerCount = 0;
+  std::string defaultGuiConfigName = "gui.config";
 
+  // The playback flag was specified from the command line
+  if (_guiConfig != nullptr && std::string(_guiConfig) == "_playback_")
+  {
+    ignition::common::env(IGN_HOMEDIR, defaultConfig);
+    defaultConfig = ignition::common::joinPaths(defaultConfig, ".ignition",
+        "gazebo", "playback_gui.config");
+    defaultGuiConfigName = "playback_gui.config";
+    auto runner = new ignition::gazebo::GuiRunner(worldsMsg.data(0));
+    runner->connect(app.get(), &ignition::gui::Application::PluginAdded, runner,
+        &ignition::gazebo::GuiRunner::OnPluginAdded);
+    ++runnerCount;
+    runner->setParent(ignition::gui::App());
```

Comment:
I think it should be possible to reduce some duplication moving some logic to the block below?

Reply:
You're right, not sure why I didn't do this before 924af51600d0083aeadfd8b682c0b9d39a157bb2